DataFormats/Provenance/src/ProductRegistry.cc

```diff
@@ -454,10 +441,12 @@ namespace edm {
     productLookup(InEvent)->setFrozen();
     productLookup(InLumi)->setFrozen();
     productLookup(InRun)->setFrozen();
+    productLookup(InProcess)->setFrozen();
 
-    transient_.eventNextIndexValue_ = productLookup(InEvent)->nextIndexValue();
-    transient_.lumiNextIndexValue_ = productLookup(InLumi)->nextIndexValue();
-    transient_.runNextIndexValue_ = productLookup(InRun)->nextIndexValue();
+    for (unsigned int i = InEvent; i < NumBranchTypes; ++i) {
```

Comment:
Again, this assumes `InEvent==0` for all time.

Reply:
Done